File: backend/app/models/document.py
from datetime import datetime
from sqlalchemy import Column, Integer, String, DateTime, JSON, Enum as SQLEnum
from sqlalchemy.sql import func
from app.core.database import Base
from app.schemas.document import DocumentStatus


class Document(Base):
    __tablename__ = "documents"

    id = Column(Integer, primary_key=True, index=True)
    filename = Column(String(255), nullable=False, index=True)
    file_path = Column(String(512), nullable=False)
    file_type = Column(String(10), nullable=False, index=True)
    file_size = Column(Integer, nullable=False)
    status = Column(
        SQLEnum(DocumentStatus),
        nullable=False,
        default=DocumentStatus.PENDING,
        index=True
    )
    meta_data = Column(JSON, nullable=True)
    config = Column(JSON, nullable=True)
    result = Column(JSON, nullable=True)

    # 时间字段
    created_at = Column(DateTime(timezone=True), server_default=func.now(), nullable=False)
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now(), nullable=False)
    processed_at = Column(DateTime(timezone=True), nullable=True)

    def __repr__(self):
        return f"<Document(id={self.id}, filename='{self.filename}', status='{self.status}')>"

File: frontend/src/types/document.ts
import type { DocumentStatus } from "@/types/commonConfig";

export interface Document {
  id: number;
  filename: string;
  file_type: string;
  file_size: number;
  file_path: string;
  status: DocumentStatus;
  doc_metadata?: Record<string, string | number | boolean>;
  created_at: string;
  updated_at: string;
  processed_at?: string;
  error_message?: string;
}

export interface LoadConfig {
  file_path: string;
  file_type: string;
  loader_name: string;
  loader_config: Record<string, string | number | boolean>;
}

export interface LangChainDocument {
  page_content: string;  // 文档内容
  metadata: {
    source?: string;     // 文档来源
    page?: number;       // 页码（如果有）
    chunk_id?: string;   // 分块ID
    [key: string]: string | number | boolean | undefined;  // 其他元数据
  };
}
